src/AggregateFunctions/AggregateFunctionIntervalLengthSum.h

```diff
@@ -144,10 +150,20 @@ class AggregateFunctionIntervalLengthSum final : public IAggregateFunctionDataHe
 
         for (size_t i = 1, sz = data.segments.size(); i < sz; ++i)
         {
-            /// Check if current interval intersect with next one then add length, otherwise advance interval end
+            /// Check if current interval intersects with next one then add length, otherwise advance interval end.
             if (cur_segment.second < data.segments[i].first)
             {
-                res += cur_segment.second - cur_segment.first;
+                if constexpr (std::is_floating_point_v<TResult>)
+                {
+                    res += cur_segment.second - cur_segment.first;
+                }
+                else
+                {
+                    TResult diff;
+                    if (!common::subOverflow(static_cast<TResult>(cur_segment.second), static_cast<TResult>(cur_segment.first), diff))
+                        res += diff;
```

Comment:
What if overflows? Shouldn't we throw an exception?
```suggestion
                    if (!common::subOverflow(static_cast<TResult>(cur_segment.second), static_cast<TResult>(cur_segment.first), diff))
                        res += diff;
                    else
                        throw DB::Exception("Decimal math overflow", ErrorCodes::DECIMAL_OVERFLOW);
```